### DISCUSSION TITLE
How to respond from fetcher if data was modified

### DISCUSSION THREAD

**Discussion by damischa1** [vercel/swr#2229] [Q&A]
Hello,

I am working on shopping cart. The shopping cart is stored in database and possibly shared with multiple users or browser windows.

I would like to implement polling database (using _refreshInterval_). There is a timestamp in database for the last modification time of the shopping cart. For optimization on the server side we could check if the cart was modified since client browser has last requested it (using _if-modified-since_ -header) and possibly responding HTTP 304 (Not modified).

But how can I signal **from the fetcher function** that the data on the server was not modified and the client should keep the old version of the cart?

I was thinking of returning previous version of the cart from the SWR cache, but how can I get reference for it in my fetcher?

**Comment by danwayne-me** [comment]
Hi, I think the HTTP Cache-control mechanism works in the browser itself, not inside the fetcher, when you send http request in your fetcher, if the server responded with a 304 code, the browser will take care of it and in the fetcher, you will already see **the response with 200 and cached data from browser**. swr will then deep compare the returned data conditionally trigger re-render.

**Reply by pip8786** [reply]
This doesn't seem to work for me, when a 304 comes back from my server, chrome just lists it as a 304 and it has no body, so I'm not sure how to return the data from the fetcher. Maybe there's a misconfiguration somewhere that I can adjust? Is there an example somewhere on how a NextJS API can be created so that the browser knows to use the data in the cache?

**Comment by pip8786** [comment]
@damischa1 Did you find a good solution for this, I feel like my problem is nearly exactly the same as yours.
